src/dft.cpp

```diff
@@ -1491,7 +1491,7 @@ std::vector<struct sourcedata> dft_fields::fourier_sourcedata(const volume &wher
         temp_struct.idx_arr.push_back(idx);
         for (size_t i = 0; i < Nfreq; ++i) {
           EH0 = dJ_weight*dJ[reduced_grid_size*i+idx_1d];
-          if (is_electric(temp_struct.near_fd_comp)) EH0 *= -1;
+          if (is_electric(temp_struct.near_fd_comp) || is_D(temp_struct.near_fd_comp)) EH0 *= -1;
```

Comment:
Would be clearer to add `is_E_or_D` (and, for symmetry, `is_H_or_B`) functions to `vec.hpp`.